src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.InProcOrRemoteHostAnalyzerRunner.cs

```diff
@@ -215,7 +215,7 @@ internal class InProcOrRemoteHostAnalyzerRunner
         // TODO: filter in OOP https://github.com/dotnet/roslyn/issues/47859
         private static ImmutableDictionary<DocumentId, ImmutableArray<DiagnosticData>> Hydrate(ImmutableArray<(DocumentId documentId, ImmutableArray<DiagnosticData> diagnostics)> diagnosticByDocument, Project project)
             => diagnosticByDocument
-                .Where(entry => project.GetTextDocument(entry.documentId)?.SupportsDiagnostics() == true)
+                .Where(entry => project.GetTextDocument(entry.documentId)?.SupportsDiagnostics() != false)
```

Comment:
It took me a minute to realize that the `null != false` case is because Source Generated documents wouldn't be returned by `GetTextDocument()`.  Are there other cases where `project.GetTextDocument()` will return `null` but we don't want to add diagnostics? Is there a cheap way to make that logic more explicit?

Reply:
Can you see if the added comment helps?